src/v/storage/tests/batch_cache_test.cc

```diff
@@ -312,3 +314,60 @@ FIXTURE_TEST(test_random_batch_sizes, batch_cache_test_fixture) {
         BOOST_REQUIRE_LE(r.waste(), max_waste);
     }
 }
+
+FIXTURE_TEST(test_mark_clean_empty, batch_cache_test_fixture) {
+    storage::batch_cache_index index(cache);
+    index.mark_clean(model::offset(1000));
+}
+
+FIXTURE_TEST(test_mark_clean_edge, batch_cache_test_fixture) {
```

Comment:
Do you think it's worth adding a randomized test that mixes dirty and clean batches across multiple ranges, that has similar checks? I guess randomization doesn't add a ton, but it would be great to have tests with multiple ranges

Reply:
https://github.com/redpanda-data/redpanda/pull/16954#discussion_r1526375262